### PR TITLE
Handle NoteOn,Vel=0 same as NoteOff

### PR DESCRIPTION
**Pull request by beau-seidon**
Updated MIDI input code to comply with MIDI specification regarding stopping playback of notes. 
A NoteOn command with Velocity = 0 should be treated identically to NoteOff.

### CODE REVIEW

**Comment by Jaffe-** [comment]
Thanks for the pull request!

The commit [dd0c5ff](https://github.com/Jaffe-/NESizer2/pull/12/commits/dd0c5ff43dd18dcca2d6a7bc03f25ea3e8741f5d) is enough to fix the issue. The other two commits introduce a new `midi_note_vel` member to `struct getvalue`, but it is not used for anything.
So before getting this in, please remove the two commits adding `midi_note_vel` to `struct getvalue` and also remove the assignment of this in the remaining commit.

We can reintroduce this member later if it is actually going to be used :)

File: src/ui/ui.h [hunk 1]
```diff
@@ -97,6 +97,7 @@ struct getvalue_config {
     } state;
     enum mode previous_mode;
     uint8_t midi_note;
+    uint8_t midi_note_vel;
```
Comment (Jaffe-) [inline src/ui/ui.h:100]: No reason to add this (yet)

File: src/ui/ui.c [hunk 2]
```diff
@@ -217,6 +217,7 @@ void getvalue_handler()
             button_led_blink(getvalue.button2);
 
         getvalue.midi_note = 0xFF;
+        getvalue.midi_note_vel = 0xFF;
```
Comment (Jaffe-) [inline src/ui/ui.c:220]: No reason to add this (yet)

File: src/midi/midi.c [hunk 3]
```diff
@@ -61,12 +61,16 @@ void midi_channel_apply(struct midi_message* msg)
     uint8_t midi_channel = get_midi_channel(msg->channel);
     switch (msg->command) {
     case MIDI_CMD_NOTE_ON:
-        if (getvalue.state == ACTIVE &&
-            getvalue.parameter.type == NOTE)
+        if (getvalue.state == ACTIVE && getvalue.parameter.type == NOTE) {
             getvalue.midi_note = msg->data1;
-        else {
+            getvalue.midi_note_vel = msg->data2;
```
Comment (Jaffe-) [inline src/midi/midi.c:66]: Not needed yet

**Comment by beau-seidon** [comment]
well i tried to re-commit it cleanly based on some threads i found on StackExchange... and i bungled it.
but at least the source is fixed, so the controller should do what it needs to but not more.

I will try to learn more about how to use github before attempting any more pull requests or commits.

**Comment by Jaffe-** [comment]
Thanks for the update!
There are several ways of going about it, but I often find it easiest to just fix it up locally and then do a git reset to remove all the previous (faulty) commits, and then making new ones. To update the PR just force-push the new commits.
In this case one commit would be sufficient. I'll squash them and merge them.

**Pull request merged**